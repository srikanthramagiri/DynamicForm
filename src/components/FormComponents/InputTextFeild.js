import React, { useState } from "react";
import { Form, Row, Col } from "react-bootstrap";
import Label from "./Label";
import AlertError from "./AlertError";
import PropTypes from "prop-types";

export default function InputTextFeild(props) {
  let { name, type, placeholder, label, changeHandler, validation } = props;
  let { regex, errorMessage } = validation ? validation : {};
  let [isvalid, setIsValid] = useState(true);
  const inputHandler = (e) => {
    setIsValid(true);
  };

  const blurHandler = (e) => {
    if (e.target.value === "") {
      setIsValid(false);
    }
    let reg = new RegExp(regex);
    setIsValid(reg.test(e.target.value));
    changeHandler(e);
  };

  return (
    <Form.Group controlId="">
      <Row>
        <Col lg={6} md={6}>
          <Label name={name} label={label} />
        </Col>
        <Col lg={6} md={6}>
          <Form.Control
            type={type}
            data-test='input-text-feild'
            name={name}
            placeholder={placeholder}
            onChange={(e) => inputHandler(e)}
            onBlur={(e) => blurHandler(e)}
            max={100}
          />
        </Col>
        <Col lg={6} />
        <Col lg={6}>
          {!isvalid ? (
            <AlertError variant={"danger"} message={errorMessage}></AlertError>
          ) : (
            ""
          )}
        </Col>
      </Row>
    </Form.Group>
  );
}

InputTextFeild.protoTypes = {
  name: PropTypes.string,
  type: PropTypes.string,
  placeholder: PropTypes.string,
  label: PropTypes.string,
  changeHandler: PropTypes.func,
  validation: PropTypes.object,
};
