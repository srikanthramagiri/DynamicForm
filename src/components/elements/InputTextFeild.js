import React, { useState } from "react";
import { Form, Row, Col } from "react-bootstrap";
import Label from "./Label";
import AlertError from "./AlertError";

export default function InputTextFeild(props) {
  let { name, type, placeholder, label, changeHandler, validation } = props;
  let { regex, errormessage } = validation ? validation : {};
  let [isvalid, setIsValid] = useState(true);
  const inputHandler = (e) => {
    setIsValid(true);
  };

  const blurHandlerValiation = (e) => {
    if (e.target.value === "") {
      setIsValid(false);
    }
    let reg = new RegExp(regex);
    setIsValid(reg.test(e.target.value));
    changeHandler(e);
  };

  return (
    <Form.Group controlId="">
      <Row>
        <Col lg={6} md={6}>
          <Label name={name} label={label} />
        </Col>
        <Col lg={6} md={6}>
          <Form.Control
            type={type}
            name={name}
            placeholder={placeholder}
            onChange={inputHandler}
            onBlur={blurHandlerValiation}
            max={100}
          />
        </Col>
        <Col lg={6}></Col>
        <Col lg={6}>
          {!isvalid ? (
            <AlertError variant={"danger"} message={errormessage}></AlertError>
          ) : (
            ""
          )}
        </Col>
      </Row>
    </Form.Group>
  );
}
