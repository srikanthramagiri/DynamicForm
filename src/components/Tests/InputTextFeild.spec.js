import { shallow } from 'enzyme';
import InputTextFeild from '../FormComponents/InputTextFeild'

describe('input text feild component', ()=> {
    const changeHandlerSpy = jest.fn();
    const wrapper = shallow(<InputTextFeild changeHandler={changeHandlerSpy}/>)
    const InputText =  wrapper.find(`[data-test='input-text-feild']`)
    
    it('should render correctly', ()=> {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).not.toBeNull()
    })
    it('should be one element', ()=> {
        expect(InputText).not.toBeNull()
    })
    it('change handler', ()=> {
        InputText.simulate('change', {target: {value: 'ho.no.1-2-3'}});
        expect(changeHandlerSpy).not.toHaveBeenCalled()
    })
    it('blur handler', ()=> {
        InputText.simulate('blur', {target: {value: ''}})
        expect(changeHandlerSpy).toHaveBeenCalled()
    })
})