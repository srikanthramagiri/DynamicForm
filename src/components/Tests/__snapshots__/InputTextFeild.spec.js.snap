// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input text feild component should render correctly 1`] = `ShallowWrapper {}`;

exports[`input text feild should render correctly 1`] = `ShallowWrapper {}`;
